dynawo/sources/Modeler/DataInterface/IIDM/DYNVscConverterInterfaceIIDM.cpp

```diff
@@ -119,6 +128,34 @@ VscConverterInterfaceIIDM::getQ() {
   return InjectorInterfaceIIDM<IIDM::VscConverterStation>::getQ();
 }
 
+double
+VscConverterInterfaceIIDM::getQMax() {
+  if (vscConverterIIDM_.has_minMaxReactiveLimits()) {
+    return vscConverterIIDM_.minMaxReactiveLimits().max();
+  } else if (vscConverterIIDM_.has_reactiveCapabilityCurve()) {
+    assert(vscConverterIIDM_.reactiveCapabilityCurve().size() > 0);
+    double qMax = 0;
+    const double pGen = - getP();
+    const IIDM::ReactiveCapabilityCurve& reactiveCurve = vscConverterIIDM_.reactiveCapabilityCurve();
+    if (pGen <= reactiveCurve[0].p) {
+      qMax = reactiveCurve[0].qmax;
+    } else if (pGen > reactiveCurve[reactiveCurve.size() - 1].p) {
+      qMax = reactiveCurve[reactiveCurve.size() - 1].qmax;
+    } else {
+      for (unsigned int i = 0; i <= reactiveCurve.size() - 2; ++i) {
+        IIDM::ReactiveCapabilityCurve::point current_point = reactiveCurve[i];
+        IIDM::ReactiveCapabilityCurve::point next_point = reactiveCurve[i+1];
+        if (current_point.p <= pGen && next_point.p >= pGen) {
+          qMax = current_point.qmax + (pGen - current_point.p) * (next_point.qmax - current_point.qmax) / (next_point.p - current_point.p);
```

Comment:
cannot we break at this point? (i.e. are the pmax/qmax sorted?)

Reply:
I am not sure to understand this point

Reply:
We need to know between which points PGen is located and then we need to do the linearization.

Reply:
I mean : the condition (current_point.p <= pGen && next_point.p >= pGen) can happen only once right? so when we have found the proper qmax cannot we break the loop?